### DISCUSSION TITLE
How can I get list of all routes?

### DISCUSSION THREAD

**Discussion by juhand777** [remix-run/react-router#10944] [Q&A]
Currently, I use useMatches to get the routes that matches with current url path. But I would like to get the list from all routes defined like useMatches returns.

**Comment by kiliman** [comment]
You should already have the full route config object since that's what is passed to `createBrowserRouter`.

It won't have everything that is included in `useMatches()` because that's based on actual routes with data, etc. But it will have the route structure.

https://reactrouter.com/en/main/routers/create-browser-router

**Reply by krasevych** [reply]
> You should already have the full route config object since that's what is passed to `createBrowserRouter`.

I dont have the full route config object how can I get list of all routes in this case?

**Comment by tatsmaki** [comment]
You can refactor your code to export your routes. Even if it is dynamically populated.

```
export const routes = [];

const Router = createBrowserRouter(routes);
```

**Comment by airtonix** [comment]
what about when we're using Framework mode and we're using flatRoutes?

**Reply by sergiodxa** [reply]
You can use `npx react-router routes --json` to get them, save that to a JSON and import it in your app or read it using `node:fs`
